fs/walk.ts

```diff
@@ -0,0 +1,66 @@
+import { FileInfo, readDirSync, statSync } from "deno";
+
+
+export interface WalkOptions {
+  maxDepth?: number;
+  exts?: string[];
+  match?: RegExp[];
+  skip?: RegExp[];
+  // FIXME don't use `any` here?
+  onError?: (err: any) => void;
+  // followSymlinks?: Boolean;
+}
+
+
+export function *walk(dir: string, options: WalkOptions = {}): IterableIterator<FileInfo> {
+  options.maxDepth -= 1;
+  let ls: FileInfo[] = [];
+  try {
+    ls = readDirSync(dir);
```

Comment:
@hayd can you add async version using `readDir`?

Reply:
Will that "just work" with iterators? I can investigate.

Reply:
I think an async version would have to produce "async iterators". See https://github.com/tc39/proposal-async-iteration

Reply:
Thanks! It seems reasonable to call these walk and walkSync respectively.